ros2/source/source/Dockerfile

```diff
@@ -1,18 +1,21 @@
 # This is an auto generated Dockerfile for ros2:source
 # generated from docker_images/create_ros2_source_image.Dockerfile.em
-FROM ubuntu:xenial
+FROM ubuntu:bionic
 
 # setup keys
-RUN apt-key adv --keyserver hkp://keyserver.ubuntu.com:80 --recv-keys 421C365BD9FF1F717815A3895523BAEEB01FA116
+# RUN apt-key adv --keyserver hkp://keyserver.ubuntu.com:80 --recv-keys 421C365BD9FF1F717815A3895523BAEEB01FA116
 
 # setup sources.list
-RUN . /etc/os-release \
-    && echo "deb http://repo.ros2.org/$ID/main $VERSION_CODENAME main" > /etc/apt/sources.list.d/ros2-latest.list
-
+# RUN . /etc/os-release \
+#     && echo "deb http://repo.ros2.org/$ID/main $VERSION_CODENAME main" > /etc/apt/sources.list.d/ros2-latest.list
+ENV DEBIAN_FRONTEND noninteractive
 # setup environment
 ENV LANG C.UTF-8
 ENV LC_ALL C.UTF-8
 
+RUN apt-get update && apt-get install -q -y tzdata \
```

Comment:
Do any packages besides `tzdata` ask for user input when they're being configured? If not I think the `ENV DEBIAN_FRONTEND noninteractive` line above could be removed by making this line `RUN DEBIAN_FRONTEND=noninteractive apt-get update && ...`

Reply:
Especially as currently dockerfile environment variables [can't just be unset](https://github.com/moby/moby/issues/3465) later by end users, and passing them inline seems to be a common [compromise](https://github.com/moby/moby/issues/9298#issuecomment-133750740).

Reply:
Yeah totally, I also have that change locally but waited for the final state of the ros2/ci PR to be merged to push all the changes to that branch

Reply:
resolved in https://github.com/osrf/docker_templates/pull/34